homeassistant/components/tedee/__init__.py

```diff
@@ -64,8 +64,9 @@ async def register_webhook() -> None:
             await coordinator.tedee_client.cleanup_webhooks_by_host(instance_url)
         except (TedeeDataUpdateException, TedeeWebhookException) as ex:
             _LOGGER.warning("Failed to cleanup Tedee webhooks by host: %s", ex)
-        webhook_url = (
-            f"{instance_url}{webhook_generate_path(entry.data[CONF_WEBHOOK_ID])}"
+
+        webhook_url = webhook_generate_url(
+            hass, entry.data[CONF_WEBHOOK_ID], prefer_external=False, allow_ip=True
```

Comment:
When getting the `instance_url` we set `allow_external` to False, but here we instead set the weaker flag `prefer_external`.

Maybe we should expose additional flags of `get_url` in `async_generate_url`?

Reply:
with the exception of HA cloud, because those should not be used with this.

Reply:
good point, exposed the extra params and switched